src/dlg/dlg.c

```diff
@@ -2,7 +2,13 @@
 // Distributed under the Boost Software License, Version 1.0.
 // See accompanying file LICENSE or copy at http://www.boost.org/LICENSE_1_0.txt
 
-#define _XOPEN_SOURCE
+#if defined(__sun)
+	#define _XOPEN_SOURCE 600
```

Comment:
Disclaimer: I'm not really familiar with this at all and I'm not aware about implications, so I leave it up to the developers.
But maybe defining `#define _XOPEN_SOURCE 600` unconditionally for all platforms would also be valid?